### PR TITLE
C51-112: Activity List Design + Functionality

### PR DESCRIPTION
**Pull request by deb1990**
## Overview
As part of this PR the Design and Functionality is implemented for the Activity List inside Activity Tab.

## How it looks
activitylist

## Technical Steps
1. Markup and styles for `ActivityCard--Big.html` was not consistent with other variants, so it has been refactored.
2. Markup and styles for `ActivityCard--Long.html` has been implemented.
3. `ActivityItemMenu.html`  has been renamed to `ActivityCard--Menu--Feed.html`
4. `ActivityFeed.js` is refactored and the markup file is named as `ActivityFeed.html`.
5. Unused code is removed from `ActivityFeed.js`.
6. Following unused files are removed, 
ActivityList.html
ActivityListItem.html
7. `ContactCard.html` is refactored to use `bootstrap popover` instead of the custom solution of before. And this resulted in significant performance benefit in the whole app.
8. Shoreditch variables related to `Form states and alerts` are added in `_variables.scss`.
9. `_activity-card--long.scss` is introduced for the styles of the Long Activity card.
10. `_activity-feed.scss` is introduced for the styles of the Activity Feed styles.
11. Styles related to the summary tab has been moved from `_case-body.scss` to `_summary-tab.scss`.
12. Shoreditch is not using `brand-success` and `brand-warning` variables for `text-success` and `text-warning`. So added `civicase__text-warning` and `civicase__text-success` for the same.

### CODE REVIEW

File: ang/civicase/ActivityCard--Long.html [hunk 1]
```diff
@@ -1 +1,131 @@
...
+        <!-- Type -->
+        <span
+          class="civicase__activity-type"
+          ng-if="activity.type !== 'File Upload'"
+          ng-class="{'civicase__activity-type--completed': (activity.category.indexOf('task') > -1 && activity.is_completed)}">
+          {{ activity.type }}
+        </span>
+        <span
+          class="civicase__activity-subject"
+          ng-if="activity.type === 'File Upload'"">
```
Comment (swastikpareek) [inline ang/civicase/ActivityCard--Long.html]: Extra `"` inverted commas

File: ang/civicase/ActivityCard--Long.html [hunk 2]
```diff
@@ -1 +1,131 @@
...
+            <li ng-if="!activity.attachments || !activity.attachments.length" class="civicase__activity-attachment-load">
+              <span>
+                <i class="civicase__activity-attachment-load-icon text-primary material-icons">loop</i> &nbsp;{{ ts('Loading...') }}
+              </span>
+            </li>
+            <li ng-repeat="file in activity.attachments || []">
+              <a class="civicase__activity-attachment__file-name"
+                title="{{ file.description || ts('Download File') }}"
+                href="{{ file.url }}">
+                <i class="civicase__activity-attachment__file-icon crm-i {{ file.icon }}"></i> &nbsp;{{ file.name }}
```
Comment (swastikpareek) [inline ang/civicase/ActivityCard--Long.html]: Any specific reason to use non breaking space?

File: ang/civicase/ActivityCard--Long.html [hunk 3]
```diff
@@ -1 +1,131 @@
...
+              </a>
+            </li>
+            <li ng-if="activity.attachments.length" role="separator" class="divider"></li>
+            <li ng-if="activity.attachments.length">
+              <a class="civicase__activity-attachment__file-name"
+                ng-href="{{ 'civicrm/case/activity/download-all-files' | civicaseCrmUrl:{ activity_id: activity.id } }}">
+                <i class="civicase__activity-attachment__file-icon crm-i fa-download"></i> &nbsp;Download all
```
Comment (swastikpareek) [inline ang/civicase/ActivityCard--Long.html]: Same as above

File: ang/civicase/ActivityCard--Long.html [hunk 4]
```diff
@@ -1 +1,131 @@
...
+        </span>
+      </div>
+      <div class="civicase__activity-card-row">
+        <!-- Subject -->
+        <div ng-if="activity.type !== 'File Upload'" class="civicase__activity-subject">{{ activity.subject }}</div>
+        <!-- End - Subject -->
+      </div>
+      <div class="civicase__activity-card-row civicase__activity-card-row--communication" ng-if="activity.category.indexOf('communication') > -1">
+        <div>
+          <span>To:&nbsp;</span>
```
Comment (swastikpareek) [inline ang/civicase/ActivityCard--Long.html]: Same as above.

File: ang/civicase/ActivityCard--Long.html [hunk 5]
```diff
@@ -1 +1,131 @@
...
+          <span ng-if="activity.target_contact_name" contact-card contacts="activity.target_contact_name"></span>
+        </div>
+        <div class="civicase__activity__right-container">
+          <span>From:&nbsp;</span>
```
Comment (swastikpareek) [inline ang/civicase/ActivityCard--Long.html]: Remove `&nbsp` from here too.

File: ang/civicase/ActivityFeed.js [hunk 6]
```diff
@@ -81,58 +53,9 @@
       getActivities(true);
     };
```
Comment (swastikpareek) [inline ang/civicase/ActivityFeed.js]: Add docs if possible

File: ang/civicase/ActivityFeed.js [hunk 7]
```diff
@@ -1,76 +1,48 @@
...
     // The ts() and hs() functions help load strings for this module.
     var ts = $scope.ts = CRM.ts('civicase');
-    var ITEMS_PER_PAGE = 25,
-      caseId = $scope.params ? $scope.params.case_id : null,
-      pageNum = 0;
+    var ITEMS_PER_PAGE = 25;
+    var caseId = $scope.params ? $scope.params.case_id : null;
+    var pageNum = 0;
+
     $scope.CRM = CRM;
```
Comment (swastikpareek) [inline ang/civicase/ActivityFeed.js]: Can we remove this if we are not using `$scope.CRM`

File: ang/civicase/ActivityFeed.js [hunk 8]
```diff
@@ -158,39 +81,71 @@
...
+        param: 'ado',
+        default: angular.extend({}, {
+          followup_nested: true,
+          overdue_first: true,
+          include_case: true
+        }, $scope.params.displayOptions || {})
+      });
+    }
+
     function groupActivities (activities) {
```
Comment (swastikpareek) [inline ang/civicase/ActivityFeed.js]: Add docs if possible

File: ang/civicase/ActivityFeed.js [hunk 9]
```diff
@@ -158,39 +81,71 @@
...
         groups.push(overdue = {key: 'overdue', title: ts('Overdue Activities'), activities: []});
       }
-      groups.push(upcoming = {key: 'upcoming', title: ts('Upcoming Activities'), activities: []});
-      groups.push(past = {key: 'past', title: ts('Past Activities'), activities: []});
+
+      groups.push(upcoming = {key: 'upcoming', title: ts('Future Activities'), activities: []});
+      groups.push(past = {key: 'past', title: ts('Past Activities - Prior to Today'), activities: []});
       _.each(activities, function (act) {
-        var activityDate = act.activity_date_time.slice(0, 10);
+        // var activityDate = act.activity_date_time.slice(0, 10);
```
Comment (swastikpareek) [inline ang/civicase/ActivityFeed.js]: Remove commented code.

File: scss/components/_utils.scss [hunk 10]
```diff
@@ -2,3 +2,11 @@
   position: relative;
   top: -1px;
 }
+
+.civicase__text-warning {
```
Comment (swastikpareek) [inline scss/components/_utils.scss]: Warning and success is already the part of the bootstrap theme. 
`text-warning` and `text-success` are the classes FYI.

File: scss/components/_summary-tab.scss [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+
+  .panel-footer {
+    border-top: 0;
+    padding: 0;
+  }
+
+  .civicase__pager {
+    border-top: 0;
+
+    paging {
```
Comment (swastikpareek) [inline scss/components/_summary-tab.scss]: Can't we add `pull-right` class to this element.

File: scss/components/_activity-feed.scss [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+  .civicase__panel-transparent-header {
+    margin: auto;
+    width: 830px;
+
+    > .panel-body {
+      background: transparent;
+      border-top: 0;
+      box-shadow: none;
+      padding: 0;
+
```
Comment (swastikpareek) [inline scss/components/_activity-feed.scss]: Remove extra spaces

**Comment by swastikpareek** [comment]
@deb1990 : Lets use the correct material icon for the 3 dot dropdown.
correct icon for the dropdown

File: scss/components/_utils.scss (hunk 10 above)
Comment (deb1990) [reply scss/components/_utils.scss]: Yes but the colors different than design. Shoreditch is not using `brand-success` and `brand-warning` variables for these two classes. forgot to mention this in the PR, apologies.

File: ang/civicase/ActivityCard--Long.html (hunk 2 above)
Comment (deb1990) [reply ang/civicase/ActivityCard--Long.html]: As discussed, its fine for now.

File: ang/civicase/ActivityCard--Long.html (hunk 3 above)
Comment (deb1990) [reply ang/civicase/ActivityCard--Long.html]: As discussed, its fine for now.

File: ang/civicase/ActivityCard--Long.html (hunk 4 above)
Comment (deb1990) [reply ang/civicase/ActivityCard--Long.html]: As discussed, its fine for now.

File: ang/civicase/ActivityCard--Long.html (hunk 5 above)
Comment (deb1990) [reply ang/civicase/ActivityCard--Long.html]: As discussed, its fine for now.

File: scss/components/_utils.scss (hunk 10 above)
Comment (swastikpareek) [reply scss/components/_utils.scss]: Oh. Yes. Np. LGTM

**Pull request merged**